File: src/Services/User/User.Domain/Models/Review.cs
namespace User.Domain;

public class Review
{
    public Guid ReviewId { get; set; }
    public string UserId { get; set; }
    public int MovieId { get; set; }
    public float Rating { get; set; }
    public string ReviewText { get; set; }
    public DateTime CreationDate { get; set; }
    public IReadOnlyCollection<Vote> Votes { get; set; } = new List<Vote>();

    public bool IsValid()
    {
        return MovieId > 0 && Rating is >= 0 and <= 10 && UserId != null && CreationDate <= DateTime.UtcNow;
    }
}

File: src/Web/couch-potatoes-web-app/src/models/review/review.ts
import Movie from '../movie';
import User from '../user';
import { Vote } from './vote';

export interface Review {
    reviewId: string;
    user: User;
    movie: Movie;
    rating: number;
    reviewText: string;
    creationDate: string;
    lastUpdatedDate: string;
    votes: Vote[];
}
